h/api/nipsa/search.py

```diff
@@ -0,0 +1,83 @@
+import copy
+
+
+def nipsa_filter(query, userid=None):
```

Comment:
It would make the rest of the module simpler to read, in my opinion, and it would save you a couple tests like asserting that this doesn't modify the query (that it never needed in the first place).